picasso/src/main/java/com/squareup/picasso3/SourceBufferingInputStream.java

```diff
@@ -33,33 +35,49 @@
   private final Buffer buffer;
   private long position;
   private long mark = -1;
+  private long markLimit = -1;
 
   SourceBufferingInputStream(BufferedSource source) {
     this.source = source;
     this.buffer = source.buffer();
   }
 
   private final Buffer temp = new Buffer();
+  // offset is the write offset in the dest array
```

Comment:
Can you move this comment inside the method so it doesn't get lost if shuffling occurs. Also please capitalize "offset" and add a period. [Comments should be sentences][1].

 [1]: https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html